### PR TITLE
MAYA-122250 UFE 3 build as Universal Binary 2 (x86_64 + arm64)

### PR DESCRIPTION
**Pull request by boudrey**
+ Change a few compiler flags to be at par with Maya
+ Fix some easy Python 3 issues.

### CODE REVIEW

**Comment by dgovil** [comment]
This will be very welcome :-)
Out of curiosity, since Qt5 doesn't have a Universal Binary 2 build, I assume you're patching it to build for both arches? Are you using the SideFX fork with the chromium backports?

A follow up question is, since Maya USD has historically required developers to provide a Qt5 install to be able to do the full build with the Qt tools, how do you see this working for devs who build maya-usd from source? Will Maya USD and the associated devkit be self sufficient for Qt in the future? (see https://github.com/Autodesk/maya-usd/issues/1070 for example of a prior request)

Anyway, this will be a very nice change to have overall! Thanks!

File: cmake/compiler_config.cmake [hunk 1]
```diff
@@ -5,8 +5,9 @@ set(GNU_CLANG_FLAGS
     # we want to be as strict as possible
     -Wall
     $<$<BOOL:${BUILD_STRICT_MODE}>:-Werror>
+    $<$<CONFIG:DEBUG>:-fstack-check>
     # optimization
-    -msse3
+    -msse4.2
```
Comment (seando-adsk) [inline cmake/compiler_config.cmake:10]: These GNU_CLANG_FLAGS are used by both clang and gcc. Is this change okay for our Linux builds?

**Review by seando-adsk — APPROVED** [review]

File: cmake/compiler_config.cmake (hunk 1 above)
Comment (seando-adsk) [reply cmake/compiler_config.cmake:10]: Yes, but current Maya is building with gcc 9, whereas we are still using gcc 6.3.1.

Comment (boudrey) [reply cmake/compiler_config.cmake:10]: This flag has been there for at least in Feb. 8, 2019 in Maya and it was merged from Peptide so it was in Peptide before that.

**Comment by boudrey** [comment]
Hi @dgovil, maybe it would be for Qt6 only. Need to start providing UB2 across all our components if we want to finally get a version of Maya with UB2.

File: cmake/compiler_config.cmake (hunk 1 above)
Comment (seando-adsk) [reply cmake/compiler_config.cmake:10]: Okay thanks for checking.

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**